src/core/server/handler/handler.cpp

```diff
@@ -0,0 +1,172 @@
+#include <handler/handler.h>
+
+handler::handler(const utility::string_t &url) : m_listener(url) {
+	m_listener.support(
+		methods::GET,
+		[this](auto &&PH1) {
+		  handle_get(std::forward<decltype(PH1)>(PH1));
+		});
+	m_listener.support(
+		methods::PUT,
+		[this](auto &&PH1) {
+		  handle_put(std::forward<decltype(PH1)>(PH1));
+		});
+	m_listener.support(
+		methods::POST,
+		[this](auto &&PH1) {
+		  handle_post(std::forward<decltype(PH1)>(PH1));
+		});
+	m_listener.support(
+		methods::DEL,
+		[this](auto &&PH1) {
+		  handle_delete(std::forward<decltype(PH1)>(PH1));
+		});
+
+}
+
+[[maybe_unused]] void handler::handle_error(const pplx::task<void> &task) {
+	try {
+		task.get();
+	}
+	catch (...) {
+		// TODO: Log the error
+	}
+}
+
+// TODO: remove this
+void display_json(
+	json::value const &jvalue,
+	utility::string_t const &prefix) {
+	ucout << prefix << jvalue.serialize() << std::endl;
+}
+
+bool authenticate(const http_headers &headers) {
+	auto auth_header = headers.find(U("authorization"));
+	std::vector<unsigned char> credentials = utility::conversions::from_base64
+		(auth_header->second.substr(6, auth_header->second.size()));
+
+	std::string username{credentials.begin(), std::find(credentials.begin(), credentials.end(), ':')};
```

Comment:
I would seriousliy prefer of this to be std::string_view.

(Not incredibly important but really nice to have.)

Reply:
It doesn't work with string_view - the constructor isn't the same